script.js

```diff
@@ -74,6 +84,26 @@
     this.message = options.message || {};
     var renderedCallback = options.onButtonRendered;
 
+    this.customSettings = options.customElementSettings || null;
+    if (this.customSettings) {
+      while (this.customElementGroup.firstChild) {
+        this.customElementGroup.removeChild(this.customElementGroup.firstChild);
+      }
+      // Focusable should be true when it is defaultFocus.
+      this.customSettings.focusable = this.customSettings.defaultFocus ?
+                                      true : this.customSettings.focusable;
+      // Set customElementOffset to 1 when custom element is focusable,
+      // note that we only support 1 custom element at the time.
+      this.customElementOffset = this.customSettings.focusable ? 1 : 0;
```

Comment:
BTW, if you use `this.customSettings.focusable |= this.customSettings.defaultFocus;`, the return value will be 1 for true and 0 for false.